torchtext/csrc/vocab.cpp

```diff
@@ -356,6 +383,49 @@ Vocab _build_vocab_from_text_file(const std::string &file_path,
   return Vocab(std::move(tokens));
 }
 
+Vocab _build_vocab_from_text_file_using_python_tokenizer(
+    const std::string &file_path, const int64_t min_freq,
+    const int64_t num_cpus, py::object tokenizer) {
+  int64_t num_lines = _infer_lines(file_path);
+  int64_t chunk_size = impl::divup(num_lines, num_cpus);
+  // Launching a thread on less lines than this likely has too much overhead.
+  chunk_size = std::max(chunk_size, GRAIN_SIZE);
+
+  std::vector<size_t> offsets;
+  impl::infer_offsets(file_path, num_lines, chunk_size, offsets);
+
+  std::vector<std::shared_ptr<IndexDict>> chunk_counters;
+
+  std::mutex m;
+  std::condition_variable cv;
+  std::atomic<int> thread_count(0);
+
+  // create threads
+  int64_t j = 0;
+  for (int64_t i = 0; i < num_lines; i += chunk_size) {
+    auto counter_ptr = std::make_shared<IndexDict>();
+    thread_count++;
+    at::launch([&, file_path, num_lines, chunk_size, j, i, counter_ptr]() {
+      parse_raw_text_file_chunk_using_python_tokenizer(
```

Comment:
This will likely cause hefty issues because of the global interpreter lock. You can't parallelize this without multiple processes.

Reply:
Ahh great catch!